### ISSUE TITLE
Is there a way to instantly update the target binary of the zig-cache folder?

### ISSUE THREAD

**Issue by heibor** [ziglang/zig#3105]
When I modified my zig source, and then I run `zig build`, the generated target file in zig-cache seems not the newest version. And when I run `zig build --cache off`, it tell me that `Unrecognized argument: --cache`. I must remove the zig-cache folder and run `zig build` again.

If I build my project by build.zig, it seems that it can not update the target binary of the zig-cache folder instantly.

**Comment by andrewrk** [comment]
Is your project open source? There are no known bugs in the caching system.

**Comment by andrewrk** [comment]
Please provide steps to reproduce the problem, and I will re-open the issue.

**Comment by heibor** [comment]
Yes, I made a mistake. My project mixed zig and C. Zig code was compiled into a static library, C code will link this static library. When I modified the zig code, the build tool did not relink this Static library, because the C code has not been modified. Now I compile the zig code into an object and add this object to the C code, so that every time the zig code is modified, the build tool can rebuild the target binary again.
